react/components/EXPERIMENTAL_TableTree/hooks/useTableTreeCheckboxes.tsx

```diff
@@ -0,0 +1,138 @@
+import { useMemo, useCallback, useEffect, useReducer } from 'react'
+import {
+  getFlat,
+  getToggledState,
+  getItemTree,
+  equalsId,
+} from './checkboxesUtils'
+
+const useTableTreeCheckboxes = ({
+  items,
+  columns,
+  onToggle,
+  childsKey = 'children',
+}: hookInput): checkboxesHookReturn => {
+  const [checkedItems, dispatch] = useReducer(reducer, [])
+
+  const itemTree = useMemo(() => {
+    return getItemTree(items, childsKey)
+  }, [items, columns])
+
+  const toggle = useCallback(
+    (item: ItemTree): void => {
+      dispatch({ type: ActionType.Toggle, itemToToggle: { item, childsKey } })
+    },
+    [checkedItems]
+  )
+
+  useEffect(() => {
+    onToggle({ checkedItems })
+  }, [toggle])
+
+  useEffect(() => {
+    const shake = (tree: ItemTree) => {
+      const childs = tree[childsKey] as Array<ItemTree>
```

Comment:
⛏ I think that this variable should be named `children`, what do you think?

Reply:
I was using `children`, but found that could mess-up the head a little, since react use the same naming for children of components.

Reply:
What do you think about naming this after `childNodes`, since we are on a `tree`? (Also, `childsKey` after `nodesKey`). 🤔

Reply:
Oh, you have a point. I agree with `childNodes`!